generator/src/main/java/io/doov/gen/ModelMapGenMojo.java

```diff
@@ -69,6 +75,12 @@
     @Parameter(required = true)
     private String packageFilter;
 
+    /**
+     * TODO find a better way to configure field path provider regarding fieldClasses & sourceClasses
+     */
+    @Parameter
+    private String fieldPathProvider;
```

Comment:
I think changing this `String` to a `List<String>` and make it work like `fieldClasses` would be enough. 

If you need to have some `sourceClasses` with a `fieldPathProvider` and not for some, you can always split it into 2 maven executions

Reply:
As-is `sourceClasses` and `fieldClasses` lists match each other using index order in the list.
A better way is to have a structure of configuration like 
```
<configuration>
    <packageFilter></packageFilter>
    <fieldClass></fieldClass>
    <sourceClass></sourceClass>
    <fieldPathProvider></fieldPathProvider>
</configuration>
```
`fieldPathProvider` being optional.

As you said different code generations can be split into separate executions.

Reply:
Yes your configuration is better, and `n` executions if needed